src/org/thoughtcrime/securesms/service/IPCAddAccountsService.java

```diff
@@ -0,0 +1,93 @@
+package org.thoughtcrime.securesms.service;
+
+import static android.content.Intent.FLAG_ACTIVITY_CLEAR_TASK;
+import static android.content.Intent.FLAG_ACTIVITY_NEW_TASK;
+
+import android.app.Service;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.os.Handler;
+import android.os.IBinder;
+import android.os.Message;
+import android.os.Messenger;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import org.thoughtcrime.securesms.ConversationListActivity;
+import org.thoughtcrime.securesms.RegistrationActivity;
+import org.thoughtcrime.securesms.connect.AccountManager;
+import org.thoughtcrime.securesms.connect.DcHelper;
+
+import java.lang.ref.WeakReference;
+
+import static org.thoughtcrime.securesms.connect.DcHelper.CONFIG_ADDRESS;
+
+import com.b44t.messenger.DcAccounts;
+import com.b44t.messenger.DcContext;
+
+/**
+ * This service is invoked by companion apps aiming to add a new account to Delta Chat
+ */
+
+public class IPCAddAccountsService extends Service {
```

Comment:
IPC means "Inter package communication"? Maye this should be commented somewhere?

Reply:
for share vs. broadcast: i would be fine with both as long as share is comparable fast and is also invisible to the user: so not UI must pop up in case there is _no_ companion app (which is probably true in >99% of cases currently) or only _one_ companion app. 

as we already that broadcasts work that way and this method is also already a little bit tested, maybe we should stay with that for now, also as we want to get this pr in soon.

if there are _really_ multiple companion apps in the future, we can think that over and also change that. so, for the list of ids where we want to broadcast to, i would not care too much about that just now.